app/models/mixins/ui_service_mixin.rb

```diff
@@ -1,24 +1,44 @@
 module UiServiceMixin
   def icons
     {
-      :ContainerReplicator => {:type => "glyph", :icon => "\uE624", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-replicator
-      :ContainerGroup      => {:type => "glyph", :icon => "\uF1B3", :fontfamily => "FontAwesome"},             # fa-cubes
-      :ContainerNode       => {:type => "glyph", :icon => "\uE621", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-container-node
-      :ContainerService    => {:type => "glyph", :icon => "\uE61E", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-service
-      :ContainerRoute      => {:type => "glyph", :icon => "\uE625", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-route
-      :Container           => {:type => "glyph", :icon => "\uF1B2", :fontfamily => "FontAwesome"},             # fa-cube
-      :Host                => {:type => "glyph", :icon => "\uE600", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-screen
-      :Vm                  => {:type => "glyph", :icon => "\uE90f", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-virtual-machine
-      :Kubernetes          => {:type => "image", :icon => provider_icon(:Kubernetes)},
-      :Openshift           => {:type => "image", :icon => provider_icon(:Openshift)},
-      :OpenshiftEnterprise => {:type => "image", :icon => provider_icon(:OpenshiftEnterprise)},
-      :Atomic              => {:type => "image", :icon => provider_icon(:Atomic)},
-      :AtomicEnterprise    => {:type => "image", :icon => provider_icon(:AtomicEnterprise)}
+      :ContainerReplicator  => {:type => "glyph", :icon => "\uE624", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-replicator
+      :ContainerGroup       => {:type => "glyph", :icon => "\uF1B3", :fontfamily => "FontAwesome"},             # fa-cubes
+      :ContainerNode        => {:type => "glyph", :icon => "\uE621", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-container-node
+      :ContainerService     => {:type => "glyph", :icon => "\uE61E", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-service
+      :ContainerRoute       => {:type => "glyph", :icon => "\uE625", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-route
+      :Container            => {:type => "glyph", :icon => "\uF1B2", :fontfamily => "FontAwesome"},             # fa-cube
+      :Host                 => {:type => "glyph", :icon => "\uE600", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-screen
+      :Vm                   => {:type => "glyph", :icon => "\uE90f", :fontfamily => "PatternFlyIcons-webfont"}, # pficon-virtual-machine
+      :MiddlewareDatasource => {:type => "glyph", :icon => "\uF1C0", :fontfamily => "FontAwesome",
+                                :css_class => "fa fa-database"},
```

Comment:
other glyphs use the approach I had in a previous version of my PR where you commented that the glyph should be taken from the decorator. However, in the decorator, it's stored in the `fa fa-database` form, hence this occurrence. It basically provides the mapping `fa fa-database` -> utf char

The reason for this was already described here https://github.com/ManageIQ/manageiq/pull/8630#discussion_r63847403